### DISCUSSION TITLE
Come gestire la traduzione dei file .dlg e .lip?

### DISCUSSION THREAD

**Discussion by crazyLillo** [tesivo/tesivo_s_VTMB_italian_translation_for_Unofficial_Patch#39] [Revisione]
ho ritradotto solo la intro dove nel teatro parla il principe, (era un sassolino nella scarpa...) così modifico il file che sta in (Unofficial_Patch\dlg\downtown la\prince1.dlg).
ma una volta fatto, a video vedevo sempre i vecchi sottotitoli, non me ne capacitavo, dato che avevo cambiato la traduzione del file dlg.
Così ho scoperto dove si annidano e iniziano i sottotitoli dell'intro, ossia in  (Unofficial_Patch\sound\character\dlg\downtown la\prince1\line1014_col_e.lip).

ho provato a modificare il file (lip), ma la mia modifica non ha effetto, anzi spariscono i sottotitoli.

così vado a guardare il file originale inglese e lo paragono con quello Italiano e noto queste piccole differenze :

originale Inglese
PHRASE unicode 40 " M y   f e l l o w   K i n d r e d , "  0.000 1.152
-------------------------------------
File italiano
PHRASE unicode 42 " M i e i   c a r i   F r a t e l l i , "  0.000 1.152

vedo quell'unicode 40 per la versione inglese e unicode 42 per quella italiana, (CHE SIGNIFICA''').

vedo anche che le parole sono spaziate in modo particolare, io ci ho provato a sovrascrivere cambiando le parole rispettando gli spazi, ma non funziona!

PERCHE???

... forse è il sonno....

**Comment by tesivo** [comment]
Ci sono sostanzialmente due posizioni da controllare per i dialoghi, \dlg e \sound\character\dlg, come appunto hai ben indagato ;)

La traduzione dei file .dlg in \dlg va svolta, come hai già fatto, tenendo presente che:
- La struttura va mantenuta. La struttura, di ogni linea si presenta nel seguente modo(se qualcuno trova eccezioni lo scriva che ne discutiamo):
`{	875	}{	(Auto-Link)	}{	(Auto-Link)	}{	881	}{	not pc.HasItem("item_w_thirtyeight")	}{	}{	}{	}{	}{	}{	}{	}{	(Auto-Link)	}`
`Numero di linea | dialogo se personaggio giocante maschile | dialogo se personaggio giocante femminile | numero di linea di destinazione | istruzioni di programmazione | vuoto | vuoto | vuoto | vuoto | vuoto | vuoto | vuoto |  frase di uscita dalla conversazione`

- Le parole tra parentesi quadre sono in realtà delle informazioni per gli animatori. Forse varrebbe la pena di lasciarle in lingua originale(alcune in passato sono state tradotte).
- (Auto-Link), (Auto-End), il numero di linea, le istruzioni di programmazione vanno chiaramente lasciati stare.
- I caratteri accentati vanno sostituiti con carattere + '. Almeno fino a quando non si trova un sistema per farglieli riconoscere.

Mentre, per quanto riguarda la traduzione dei file .lip in \sound\character\dlg:
- Ci interessa solo la riga tra parentesi graffe nella sottosezione english di CLOSECAPTION
- "unicode" diventa "char"
- La stringa tra doppi apici va tradotta evitando i caratteri accentati.
- il numero che segue unicode deve essere sostituito con il numero di caratteri della stringa tra doppi apici + 1 (perchè la stringa chiude con un carattere "invisibile" di terminazione)

Per esempio, nel tuo caso la traduzione sarebbe:
`PHRASE char 22 "Miei cari Fratelli, " 0.000 1.152`
Nel conteggio dei caratteri conta anche lo spazio vuoto.

**Comment by tesivo** [comment]
Aggiungo che:
in futuro potremmo riuscire ad integrare i caratteri accentati. Potremmo riuscire a utilizzare direttamente unicode. potremmo sostituire anche english con italian, così da avere l'italiano senza sostituire la lingua inglese.

**Reply by NarakuITA** [reply]
Sarebbe veramente grandioso! Così da avere una traduzione indipendente, che non sovrascrive nulla e con un suo installer!

**Reply by crazyLillo** [reply]
Che ci sono due posizioni, l'ho ormai capito, e ho capito anche che tutto quello che ho tradotto lo dovro 

ritrascrivere su altri file.......lasciatemelo dire (...##@òqxxzhky%$£?'...) tradotto alla buona= (li mortacci 

de pippo).

per quanto riguarda i file dlg, io li apro (gli originali) togliendo poi la spunta a (a capo automatico) in 

questo modo riesco a mantenere la struttura inalterata, correggendo in tempo reale in caso occorresse, e 

tutto fila liscio.
invece per le frasi nelle parentesi quadre, io le lascierei, traducendole anche. Male non fa, sono sempre 

indicazioni utili per un futuro doppiaggio, difatti a quello sono servite.

l'ho sempre saputo dell'annoso problema dei caratteri accentati, è sempre esistito, mi ricordo che 

moltissimo tempo fa c'era già questo problema, appunto per colpa della lingua che si traduceva che non 

aveva lettere accentate nell'alfabeto.
Un sistema per fargliele riconoscere dici?, bha! ho i miei dubbi, se non l'hanno risolto finora, bhoooo.
a proposito ricontrollerò il file da me prodotto che mi sa che ho lasciato qualche "E" maiuscola 

accentata, non lo so, non sono sicuro, ma controllerò. tanto è una cosa velocissima.

NO, non mi è chiaro , Tu scrivi:

Mentre, per quanto riguarda la traduzione dei file .lip in \sound\character\dlg:
 Ci interessa solo la riga tra parentesi graffe nella sottosezione english di CLOSECAPTION "unicode" 

diventa "char" (((NON MI è CHIARO)))
(ho fatto il paragone con il file originale e l'unica cosa diversa era unicode 40 per inglese e unicode 42 

per italiano.
cosa intendi  per "unicode" diventa "char".???

Tu scrivi:

il numero che segue unicode deve essere sostituito con il numero di caratteri della stringa tra doppi apici 

più 1 (perchè la stringa chiude con un carattere "invisibile" di terminazione)
Per esempio, nel tuo caso la traduzione sarebbe:
PHRASE char 22 "Miei cari Fratelli, " 0.000 1.152
Nel conteggio dei caratteri conta anche lo spazio vuoto.


hei!, io conto 20 caratteri compresi gli spazi, più 1 sono 21 !!!

non è chiarissimo, ma mi hai aperto un po gli occhi comunque, grazie
infatti non potevo vedere ciò che era invisibile, ma tu mi hai dato la dritta giusta.


facendo un paragone approfondito (sapendo adesso del carattere invisibile di terminazione) mi sa che ho 

capito come funzia, mi serve solo conferma. seguitemi nella mia divagazione:

 **Attenzione tenere conto che qui sul forum la formattazione originale sparisce
e gli spazi non sembrano quelli che sono veramente in raltà**

originale Inglese
PHRASE unicode 40 " M y   f e l l o w   K i n d r e d , "
-------------------------------------
File italiano
PHRASE unicode 42 " M i e i   c a r i   F r a t e l l i , "

Dalla mia analisi  ho notato che :
1) - Le singole lettere che compongono una parola sono distanziate da 1 spazio. (la ragione presumo sia 
per una piu facile lettura dei sottotitoli a schermo)
2) - ogni parola è distanziata dall'altra di 3 spazi.
3) - I caratteri Invisibili di terminazione risultano essere 3

infatti contando fra gli apici --->  originale Inglese (unicode 40 " M y   f e l l o w   K i n d r e d , ".   

Risultano 37 caratteri compresi gli spazi vuoti, più 3 invisibili = 40

per quello Italiano lo stesso meccanismo a quanto pare, ditemi voi se sbaglio.

infatti contando fra gli apici ---> File italiano unicode 42 " M i e i   c a r i   F r a t e l l i , ". Risultano 39 

caratteri compresi gli spazi vuoti, più 3 invisibili = 42

mi sembra che adesso tutto torni
Adesso però vorrei sapere se mi sono bevuto il cervello oppure è veramente come ho esposto qui 

sopra???
ancora non ho avuto tempo di provarci, lo farò a breve e scoprirò...

----------------------------

NarakuITA scrive:

Sarebbe veramente grandioso! Così da avere una traduzione indipendente, che non sovrascrive nulla e 

con un suo installer!

in pratica lavorando alla versione inglese, come ho fatto io, accorgendomi di frasi mancanti, si inizia a 

portarla di fatto alla 10.7 versione Italiana. Vai ragazzi!

**Reply by NarakuITA** [reply]
Quando vuoi mantenere la formattazione originale, per mostrare tutto il comando con il testo, utilizza la funzione: Insert Code. Sarebbe l'opzione con l'icona <>.
Esempio con la funzione: `PHRASE char 22 "Miei cari Fratelli, " 0.000 1.152`
Senza la funzione: PHRASE char 22 "Miei cari Fratelli, " 0.000 1.152

Per il resto, la risposta tecnica la conosce meglio tesivo.

**Reply by crazyLillo** [reply]
si, ho capito, adesso anche ricordo, (ho una reminescenza da forum di un millennio fa).
si,si, per mostrare la formattazzione originale. ok. grazie sempre delle dritte.

PERO'

aiuto! non ci siamo, sono bloccato.

non riesco a fargli accettare le modifiche, ho provato in diversi modi.
ma niente, i sottotitoli della frase vengono sostituiti da punti interrogativi.


```
originale Inglese
PHRASE unicode 40 " M y   f e l l o w   K i n d r e d , "  0.000 1.152
-------------------------------------
File italiano
PHRASE unicode 42 " M i e i   c a r i   F r a t e l l i , "  0.000 1.152
```


addirittura ho provato a cambiare una singola lettera e ho messo "zari" al posto di "cari",
scambiando la "c" con "z". ho cambiato un singolo carattere e non ho spostato ho aggiunto nient'alro.
ho salvato e niente non funziona! come è possibile???
devo rimettere il file di backup per farlo rifunzionare.
e che cax!

per favore fatemi capire come fare veramente sennò non posso andare avanti.

nota : io apro il file con blocco note e so come aprirli

magari mi sfugge qualcosa che ho sotto gli occhi, boooo.

**Reply by NarakuITA** [reply]
Usa sempre Notepad++
[https://notepad-plus-plus.org/downloads/](url)

**Reply by tesivo** [reply]
Se si vuole tradure anche le frasi tra parentesi quadre va bene, semplicemente è un aspetto secondario

Per i caratteri accentati: se si lascia un carattere accentato semplicemente si vedrà un simbolo sbagliato al suo posto, quindi niente di grave. Ho provato ad utilizzare "ascii" al posto di char/unicode, ma non ho avuto fortuna. Mi sembra però che i caratteri speciali nella traduzione francese funzionino correttamente quindi un sistema per risolvere deve esistere. Comunque è un problema molto secondario di cui non mi preoccuperei troppo. Anche qui se in futuro si troverà una soluzione ci si mette un secondo a riprendere i caratteri accentati usando l'apostrofo e a rimetterli con l'accento.

Si, anche le spaziature contano come caratteri. Diverso è il caso unicode dove gli zeri non contano.

Si, notepad++ è ottimo, oppure va bene anche un ide tipo visual studio/pycharm

**Comment by crazyLillo** [comment]
ho provato con notepad è questa volta qualcosina è cambiata, ma sono ancora in alto mare.
allora ho aperto il file in questione, è questa sotto è l'immagine.

![1](https://user-images.githubusercontent.com/96891196/149044180-27d46c6f-7323-4756-9e7f-556b1c2a6f34.jpg)

**e l'ho modificata così**

![2](https://user-images.githubusercontent.com/96891196/149044267-b143668e-c89d-48da-9dc5-d02686ca06a8.jpg)


Adesso a video si vede solo "Miei fratelli ????"

in notepad vedo che c'è quel carattere "NUL", io al posto di quello ho messo spazi.


ma insomma

**Reply by tesivo** [reply]
Rispondo qui e poi recupero le risspote precedenti. [recuperate e risposto]

Puoi usare l'editor che più ti aggrada(se lavori in char): notepad, wordpad, notepad++, l'editor di un IDE di programmazione, quello che preferisci.

Il file non funziona perchè hai fatto un misto tra unicode e char, dove solo la parte unicode verrà letta dal gioco.
O lavori in unicode anche per la scritta seguaci (devi in questo caso usare un editor esadecimale o notepad++ con il plugin per l'hex editor) oppure, consigliato, rendi tutto char e scrivi la frase normalmente, senza spazi che non servono, senza null. Così:

`PHRASE char 24 "Miei Fratelli seguaci." 0.000 1.152`

**Reply by crazyLillo** [reply]
ragazzi, scusate se ho alzato un polverone con il mio sos.
a volte mi perdo in un bicchier d'acqua...

ieri notte ci sono arrivato per vie traverse, ho utilizzato un editor esadecimale per accorgermi che quei caratteri "Nul" altro non erano che "zeri" e non spazi normali.
così li ho inseriti nei posti giusti aggiungendo i 3 caratteri di terminazione alla voce "unicode" sempre usando l'editor esadecimale.

Funziona!

poi ho fatto una ricerca per vedere come inserire i caratteri "Nul" su notepad e allora ho capito una volta per tutte.


ma adesso che leggo la tua risposta, mi è ancora più chiaro.
infatti, qualche messaggio indietro, qui da qualche parte, avevo chiesto che significasse  cambiare in "char".

adesso penso di aver capito, basta cambiare la scritta "unicode" con "char", è giusto?
e poi rispettare la numerazione dei caratteri immessi, ma non ho capito bene i caratteri di terminazioni invisibili quanti ne devono essere.
ho visto che per unicode, sono 3, invece per "char" sono 2.

spiegatemi bene sto passaggio dei caratteri di terminazione invisibili per favore

quindi

`PHRASE unicode 48 " M i e i   F r a t e l l i   s e g u a c i . " 0.000 1.152`

diventa

`PHRASE char 24 "Miei Fratelli seguaci." 0.000 1.152`


ok, ma credo di aver capito del perchè degli spazi fra le singole lettere!

senza spazi credo che siano meno visibili a schermo, non lo so,dovrò fare delle prove.
la domanda è.....perchè le hanno spaziate in quel modo?
anche nella versione originale inglese sono spaziate!

**Reply by tesivo** [reply]
Si, cambi la scritta "unicode" con "char". No, per char è 1 il carattere di terminazione, anche se rappresentato con 2 caratteri è considerato 1.

La frase è giusta. Forse hai ragione, gli spazi servono per aumentare la leggibilità del testo, forse c'è un altro motivo, boh :D comunque per quello che vogliamo fare noi va benissimo con char. Tanto poi per tornare al sistema unicode possiamo utilizzare un script che aggiunge gli spazi (gli zeri), cambia char con unicode e ricalcola il numero di caratteri.

**Comment by crazyLillo** [comment]
fatto, ho ritradotto l'intro e altro

**Reply by tesivo** [reply]
Perfetto ;)

**Comment by crazyLillo** [comment]
sto diventando un po più esperto ogni giorno...


Nel SDK ho trovato questo ottimo programmino che hanno fatto in cecoslovacchia per la localizzazione del gioco
.

Ho rifatto la traduzione su questo software, usando il file originale inglese, differenziando tutte le risposte (male\female).
su notepad si potrebbe anche sbagliare qualcosa per una svista o altro.
qui non è possibile sbagliare perchè si va a toccare solo il testo.
un ottimo programma, le cose sono molto facilitate, ma c'è da lavorare lo stesso.


![bde](https://user-images.githubusercontent.com/96891196/150806609-a8927701-68a7-440c-90dd-cf65fb4806a6.jpg)

**Reply by tesivo** [reply]
Davvero un programma interessante!

**Reply by crazyLillo** [reply]
Un programma più che ottimo, non immagini quanto, i ragazzi della cecoslovacchia sapevano quello che facevano.

Non avevo dato una guardata all'help del programma, perchè ho trovato l'uso di una semplicità disarmante.

ma dopo averlo letto, ho capito molte cose.

L'help è solo in inglese e non si puo fare copia e incolla per leggerlo con calma.

così l'ho trascritto su blocco note e l'ho tradotto in Italiano, a qualcuno potrebbe servire se approdasse qui e volesse cimentarsi.

ecco la guida in Italiano:

INTRODUZIONE

Questo è un sofisticato editor di dialoghi per Vampire the Masquerade: Bloodlines.
Accetta file di tipo DLG che fanno parte del gioco. Si noti che alcuni degli originali
I file DLG hanno una struttura danneggiata che è necessario riparare manualmente. Quando si apre un
file rotto, il messaggio di errore ti dirà il numero della linea che deve essere riparata -
di solito manca il numero di riga o la parentesi graffa.

Il lavoro con l'editor può essere estremamente veloce una volta che impari ad usarlo. 
La chiave è prendere confidenza  di tutti i tasti di scelta rapida.

Note

* Le linee maschili e femminili sono su sfondo verde se sono identiche e su sfondo rosso se differiscono.

* se hai un filtro attivo, la finestra principale verrà visualizzata su sfondo blu.

* Il filtro Auto-end mostra le righe che l'NPC dice dopo il dialogo vero e proprio in cui sono presenti i sottotitoli
che appaiono nell'angolo in alto a sinistra dello schermo nel gioco. Questi sottotitoli devono essere modificati
nei File LIP che puoi trovare nella directory "sound". Il numero di riga è una parte del nome di un file LIP.


* Il filtro Differenza mostra le righe per l'NPC in cui i campi Maschio e Femmina differiscono. Questo
è importante se vuoi ascoltare il suono di diverse linee femminili nel gioco. Hai bisogno
di annotare i numeri di riga, trovare i rispettivi file nella directory sound e copiare l'intera terzina
di MP3, LIP e VCD dalla variante " col_e to col_f ". Quindi se il filtro Differenza ti dice che
la riga 61 per Vandal è una da copiare, 
si individuano i file ../sound/character/dlg/santa monica/vandal/line_61_col_e. [mp3,lip,vcd] 
e copiali in line_61_col_f.[mp3.lip.vcd] nella stessa directory. 
Devi estrarre tutti i dati dai pacchetti in anticipo però.

* Il filtro personalizzato ti consente di visualizzare solo le righe che soddisfano il tuo criterio. È utile se vuoi
per verificare la presenza di un errore di battitura che hai duplicato più volte durante il lavoro o per scoprire tutte
le linee che portano a un'altra linea del dialogo.

* La colonna " * " nella finestra principale indica le righe che sono state modificate da quando
il file era aperto. Le righe modificate e non salvate vengono visualizzate in rosso nella finestra principale.

* Puoi riorganizzare la larghezza delle colonne nella finestra principale come meglio credi.
l'applicazione ricorderà l'impostazione.

* Quando si passa da una visuale filtro all'altra, l'applicazione ricorda il numero di riga.

HOTKEYS

```
Ctrl+O                           Open file
Ctrl+W                          Close file
Ctrl+S                           Save file
Ctrl+Shift+S                  Save as...
Ctrl+Q                           Quit application
Ctrl+T                           Go to line...
Ctrl+G                          Go to next line in diologue from the current line
Tab                               Cycles among Main window and three input fields 
Enter                           Switches the focus from current input field to main window or back
Ctrl+D                        Duplicates Male line input field to Female line input field
Ctrl+M                        Duplicates Male line input field to Malkavian line input field
Ctrl+F                       Finds a search string in the whole file (can find line numbers)
Ctrl+R                      Replicates to current line's input fields the contents of respective
                                 fields on the line you enter

Ctrl+1                        Cancels all filter (activates All items filter)
Ctrl+2                        Activates Auto-end filter
Ctrl+3                        Activates Difference filter
Ctrl+4                        Activates Non-empty lines filter
Ctrl+5                        Activates last custom filter
Navigation Keys         In Main window they scroll over the list of lines
Alt+Tab                      In input field switches from Edited to Original view of line or back
```

-----------------------------------------------------------

**Reply by NarakuITA** [reply]
Questo si che è un ottimo programma, ma ottima scoperta!

**Reply by crazyLillo** [reply]
hi hi hi. ho scoperto l'acqua calda troppo tardi!

Era li nella cartella (patch extra) dell'unofficial patch da diversi anni ormai...

Veramente un ottimo programma, ottimo per differenziare il male/female,
e come scritto nella guida che ho tradotto, ti spiega come ricreare linee complete al femminile.

ho scoperto il perchè del casino male/female, ma ora è lunga da dire, 
ma in finale è colpa anche di come fu rilasciato il gioco al tempo
